### PR TITLE
Extract Judge0 code execution engine as module

### PR DESCRIPTION
**Pull request by anunnakian**
Please see the issue https://github.com/langchain4j/langchain4j/issues/1048

## Change
I extract these classes as new module `langchain4j-code-execution-engine-judge0` : 
- `Judge0JavaScriptEngine`
- `JavaScriptCodeFixer`
- `Judge0JavaScriptExecutionTool`
- `JavaScriptCodeFixerTest`

and I moved the `com.squareup.okhttp3:okhttp` dependency from the main module to that new one.

## General checklist

- [x] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [x] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [x] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)

## Checklist for adding new model integration

- [x] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for adding new embedding store integration

- [x] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [x] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for changing existing embedding store integration

- [x] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j/pom.xml [hunk 1]
```diff
@@ -23,8 +23,8 @@
         </dependency>
 
         <dependency>
-            <groupId>com.squareup.okhttp3</groupId>
-            <artifactId>okhttp</artifactId>
+            <groupId>com.squareup.retrofit2</groupId>
```
Comment (langchain4j) [inline langchain4j/pom.xml]: Is retrofit required for something in `langchain4j` module? The idea of #1048  was to get rid of all okhttp/retrofit dependencies

**Review by ghost — COMMENTED** [review]
@anunnakian thanks a lot!

File: langchain4j/pom.xml (hunk 1 above)
Comment (anunnakian) [reply langchain4j/pom.xml]: Yes, fixed ;)

**Comment by langchain4j** [comment]
@anunnakian thank you a lot!

**Comment by konczdev** [comment]
@langchain4j I think it is late but the base package should be `dev.langchain4j.code.judge` or `dev.langchain4j.code.judge0` instead of `dev.langchain4j.code`. 

The [langchain4j-code-execution-engine-graalvm-polyglot](https://github.com/langchain4j/langchain4j/tree/main/code-execution-engines/langchain4j-code-execution-engine-graalvm-polyglot) did this correctly with `dev.langchain4j.code.graalvm`.

Keeping `dev.langchain4j.code` introduces a new split package problem with [langchain4j-core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core). #1066 

Edit: maybe the checklist should include the base package uniqeness for each module

**Comment by anunnakian** [comment]
Yes you're right, my bad!
I'll fix this

**Comment by anunnakian** [comment]
Here is the fix @konczdev : https://github.com/langchain4j/langchain4j/pull/1136 
@langchain4j what do you think ?

**Comment by konczdev** [comment]
@anunnakian looks good. 

There are 2 references in the the example repo too: 
[tutorials/src/main/java/_11_ServiceWithDynamicToolsExample.java](https://github.com/langchain4j/langchain4j-examples/blob/main/tutorials/src/main/java/_11_ServiceWithDynamicToolsExample.java)
[other-examples/src/main/java/ServiceWithDynamicToolsExample.java](https://github.com/langchain4j/langchain4j-examples/blob/main/other-examples/src/main/java/ServiceWithDynamicToolsExample.java)

**Comment by anunnakian** [comment]
I'll update that after the merge, Thank you @konczdev

**Comment by langchain4j** [comment]
@konczdev thanks a lot for spotting it and @anunnakian for fixing!

**Pull request merged**